### ISSUE TITLE
When child class delegates `to_s` to parent, JRuby outputs child object attributes

### ISSUE THREAD

**Issue by philipmw** [jruby/jruby#1226]
Evaluate this:

```
class BaseBlah; end
class Blah < BaseBlah
  def initialize
    @my_attr = {secret: 'sec'}
  end
  def to_s
    super.to_s
  end
end
puts "#{[Blah.new].to_s}"
```

In Ruby MRI, output:

```
[#<Blah:0x002b25ba7e2588>]
```

In JRuby 1.7.7, output:

```
[#<Blah:0x6c5739ec @my_attr={:secret=>"sec"}>]
```

**Comment by enebo** [comment]
JRuby 9.x is correct and JRuby 1.7.x is wrong in 1.9 mode.

**Comment by enebo** [comment]
This works in 9k and we will never fix this in 1.7.x as it is winding down....resolving.
